src/vsr/superblock.zig

```diff
@@ -662,21 +688,33 @@ pub fn SuperBlockType(comptime Storage: type) type {
             superblock.acquire(context);
         }
 
+        const UpdateCheckpoint = struct {
+            commit_min_checksum: u128,
+            commit_min: u64,
+            commit_max: u64,
+        };
+
         /// The vsr_state must update the commit_min and commit_min_checksum.
-        // TODO Will the replica ever update view/log_view by calling checkpoint() during a view
-        // change? If not, forbid it.
+        /// The vsr_state must not update view/log_view.
         pub fn checkpoint(
             superblock: *SuperBlock,
             callback: fn (context: *Context) void,
             context: *Context,
-            vsr_state: SuperBlockSector.VSRState,
+            update: UpdateCheckpoint,
         ) void {
```

Comment:
Previously `SuperBlock.view_change()` and `SuperBlock.checkpoint()` took a `VSRState` parameter, and then they did a bunch of assertions to make sure that only the correct fields were updated. (`checkpoint` updates commits, `view_change` updates view numbers). Now only the parameters that _can_ be modified are passed in.